File: 1019-next-greater-node-in-linked-list/1019-next-greater-node-in-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int[] nextLargerNodes(ListNode head) {
        int n = 0;
        ListNode temp = head;
        
        while(temp != null) {
            temp = temp.next;
            n++;
        }
        
        int[] arr = new int[n];
        temp = head;
        ListNode temp2; 
        int index = 0;
        
        while(temp.next != null) {
            boolean isFound = false;
            temp2 = temp;
            
            while(temp2 != null) {
                if(temp.val < temp2.val) {
                    arr[index] = temp2.val;
                    isFound = true;
                    break;
                }
                
                temp2 = temp2.next;
            }
    
            if(isFound == false)
                arr[index] = 0;
            index++;
            temp = temp.next;
        }
        
        return arr;
    }
}

File: 1019-next-greater-node-in-linked-list/1019-next-greater-node-in-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def nextLargerNodes(self, head: Optional[ListNode]) -> List[int]:
        res = []
        stack = []
        index = 0
        
        while head:
            while stack and stack[-1][1] < head.val:
                removed = stack.pop()
                res[removed[0]] = head.val
            
            res.append(0)
            stack.append((index, head.val))
            head = head.next
            index += 1
            
        return res